### ISSUE TITLE
Unresolved symbol in tablecloth/let-dataset function

### ISSUE THREAD

**Issue by radovanne** [clj-kondo/clj-kondo#2371]
- [x] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**version**
clj-kondo v2024.08.01

**platform**
native version

**editor**
editor: neovim
plugin: conjure and lsp

**problem**

I am not sure if this would be a bug or feature request...
I am using [scicloj/tablecloth](https://github.com/scicloj/tablecloth) and when using let-dataset func clj-kondo is showing "Unresolved symbol", but I am able to evaluate the code and it works like intended.
`(tc/let-dataset [x (range 1 6)
                          y 1
                          z (dfn/+ x y)])`
Linter is saying:
`clj-kondo: Unresolved symbol: x [unresolved-symbol]
clj-kondo: Unresolved symbol: y [unresolved-symbol]
clj-kondo: Unresolved symbol: z [unresolved-symbol]`

**repro**
deps:
`
{:deps {org.scicloj/clay {:mvn/version "2-beta15"}
        techascent/tech.ml.dataset {:mvn/version "7.030"}
        scicloj/tablecloth {:mvn/version "7.029.2"}}}
`
`(ns index
  (:require [scicloj.clay.v2.api]
            [tablecloth.api :as tc]
            [tech.v3.datatype.functional :as dfn]))

(tc/let-dataset [x (range 1 6)
                 y 1
                 z (dfn/+ x y)])
`
**expected behavior**

Treat let-dataset func as normal clojure let binding without showing unresolved simbol.

![image](https://github.com/user-attachments/assets/e52f670e-832c-4323-be3d-2f810ddc0b7a)

**Comment by borkdude** [comment]
This is not a bug but something that can be solved through configuration. 
Clj-kondo is a static analyzer and doesn't know about custom macros.
It seems to be that the syntax of this macro is similar to clojure.core/let.
So in `.clj-kondo/config.edn` you say put:

```
{:lint-as {foo.bar/custom-let clojure.core/let}}
```

(of course change the namespace and var name to what you are using, fully qualified).

You can check https://github.com/clj-kondo/clj-kondo/blob/master/doc/config.md#unrecognized-macros for more info.

**Comment by radovanne** [comment]
Thank you for response and for the link!
I guess that issue could be solved by applying metadata to macro in the source code, that way it will be linting correctly to everyone who uses that macro.

**Comment by borkdude** [comment]
Yes, or in general with https://github.com/clj-kondo/clj-kondo/blob/master/doc/config.md#exporting-and-importing-configuration
